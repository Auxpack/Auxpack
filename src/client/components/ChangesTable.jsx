import React from 'react';
import PropTypes from 'prop-types';
import Table from '@material-ui/core/Table';
import { makeStyles } from '@material-ui/core/styles';

const ChangesTable = props => {
    return <div>
        <h3>Changes from Previous Build</h3>
    </div>
}

export default ChangesTable;