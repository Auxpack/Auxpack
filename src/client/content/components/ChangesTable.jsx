import React from 'react';
import { makeStyles } from '@material-ui/core/styles';
import ExpansionPanel from '@material-ui/core/ExpansionPanel';
import ExpansionPanelSummary from '@material-ui/core/ExpansionPanelSummary';
import ExpansionPanelDetails from '@material-ui/core/ExpansionPanelDetails';
import Typography from '@material-ui/core/Typography';
import ExpandMoreIcon from '@material-ui/icons/ExpandMore';

const useStyles = makeStyles(theme => ({
    root: {
        width: '100%',
    },
    heading: {
        fontSize: theme.typography.pxToRem(15),
        fontWeight: theme.typography.fontWeightRegular,
    },
}));

const ChangesTable = props => {
    const { dirFinalArrayPrev, dirFinalArray, getBytes } = props;
    // console.log(`prev: `, props.dirFinalArrPrev, `current: `, dirFinalArray)

    // Changes filtering
    const dirFinalFiles = [];
    for (let i = 0; i < dirFinalArray.length; i++) {
        for (let j = 1; j < dirFinalArray[i].length; j++) {
            for (let k = 0; k < dirFinalArray[i][j].length; k++) {
                dirFinalFiles.push([dirFinalArray[i][0] + '/' + dirFinalArray[i][j][k].filename, dirFinalArray[i][j][k].size])
            }
        }
    }

    const dirFinalFilesPrev = [];
    for (let i = 0; i < dirFinalArrayPrev.length; i++) {
        for (let j = 1; j < dirFinalArrayPrev[i].length; j++) {
            for (let k = 0; k < dirFinalArrayPrev[i][j].length; k++) {
                dirFinalFilesPrev.push([dirFinalArrayPrev[i][0] + '/' + dirFinalArrayPrev[i][j][k].filename, dirFinalArrayPrev[i][j][k].size])
            }
        }
    }

    // added files
    const added = dirFinalFiles.filter((curr) => {
        let previous = dirFinalFilesPrev.map((item) => item[0])
        return !previous.includes(curr[0])
    })
    // removed files
    const removed = dirFinalFilesPrev.filter((curr) => {
        let original = dirFinalFiles.map((item) => item[0])
        return !original.includes(curr[0])
    });
    const additions = [];
    if (added.length === 0) {
        additions.push({ path: 'No additions', size: 0 })
    } else {
        for (let i = 0; i < added.length; i += 1) {
            const path = added[i][0]
            const size = added[i][1]
            additions.push({ path, size });
        }
    }
    const removals = [];
    if (removed.length === 0) {
        removals.push({ path: 'No removals', size: 0 })
    } else {
        for (let i = 0; i < removed.length; i += 1) {
            const path = removed[i][0]
            const size = removed[i][1]
            removals.push({ path, size });
        }
    }

    const additionListItems = additions.map((obj, i) => {
        return (<tr key={i} className="table-row">
            <td>{obj.path}</td>
            <td>{getBytes(obj.size)}</td>
        </tr>)
    })
    const removalListItems = additions.map((obj, i) => {
        return (<tr key={i} className="table-row">
            <td>{obj.path}</td>
            <td>{getBytes(obj.size)}</td>
        </tr>)
    })

    const AdditionCard = () => {
        return (<table className="highlight">
            <thead>
                <tr className="card-body">
                    <th>File Path</th>
                    <th>File Size</th>
                </tr>
            </thead>
            <tbody>
                {additionListItems}
            </tbody>
        </table >)

    }

    const RemovalCard = () => {
        return (
            <table className="highlight">
                <thead>
                    <tr className="card-body">
                        <th>File Path</th>
                        <th>File Size</th>
                    </tr>
                </thead>
                <tbody>
                    {removalListItems}
                </tbody>
            </table >

        )
    }

    const SimpleExpansionPanel = () => {
        const classes = useStyles();

        return (
            <div className={classes.root} >
                <ExpansionPanel className="expansionPanel">
                    <ExpansionPanelSummary
                        expandIcon={<ExpandMoreIcon />}
                        aria-controls="panel1a-content"
                        id="panel1a-header"
                    >
                        <Typography className={classes.heading} className="expansion-heading">
                            {/* Expansion heading */}
                            <strong className="centered">Additions</strong>
                        </Typography>
                    </ExpansionPanelSummary>
                    <ExpansionPanelDetails className="panelDetails">
                        {/* Additions Card Panel - content*/}
                        <AdditionCard />
                    </ExpansionPanelDetails>
                </ExpansionPanel>
                {/* Second expansion */}
                <ExpansionPanel className="expansionPanel">
                    <ExpansionPanelSummary
                        expandIcon={<ExpandMoreIcon />}
                        aria-controls="panel1a-content"
                        id="panel1a-header"

                    >
                        {/* Expansion heading */}
                        <Typography className={classes.heading} className="center-heading" className="expansion-heading">
                            <strong className="centered">Removals</strong>
                        </Typography>
                    </ExpansionPanelSummary>
                    <ExpansionPanelDetails className="panelDetails">
                        {/* Removals Card Panel - content*/}
                        <RemovalCard />
                    </ExpansionPanelDetails>
                </ExpansionPanel>

            </div>
        );
    }

    return <div className="cards-container centered">
        <SimpleExpansionPanel />
    </div>
}

export default ChangesTable;
