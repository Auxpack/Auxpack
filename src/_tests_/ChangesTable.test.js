import ChangesTable from '../client/content/components/ChangesTable.jsx';

describe('ChangesTable Unit Tests', () => {
  let wrapper;

  const props = {
    dirFinalArrayPrev: [
      ['client',
        { filename: 'index.js', size: 164, percentage: '0.01%' },
      ],
      ['client/Components',
        { filename: 'AnimalCard.jsx', size: 795, percentage: '0.07%' },
        { filename: 'AnimalDisplay.jsx', size: 743, percentage: '0.06%' },
        { filename: 'Form.jsx', size: 2008, percentage: '0.17%' },
        { filename: 'List.jsx', size: 3910, percentage: '0.33%' },
        { filename: 'ListItem.jsx', size: 792, percentage: '0.07%' },
      ],
      [
        'client/stylesheets',
        [{ filename: 'styles.css', size: 453, percentage: '0.04%' }],
      ],
      [
        'node_modules/css-loader/dist/cjs.js!./node_modules/sass-loader/dist/cjs.js!./client/stylesheets',
        {
          filename: 'styles.css',
          size: 1293,
          percentage: '0.11%',
        },
      ],
      [
        'node_modules/css-loader/dist/runtime',
        { filename: 'api.js', size: 2677, percentage: '0.23%' },
      ],
    ],
    dirFinalArray: [
      ['client',
        { filename: 'index.js', size: 164, percentage: '0.01%' },
      ],
      ['client/Components',
        { filename: 'AnimalCard.jsx', size: 795, percentage: '0.07%' },
        { filename: 'AnimalDisplay.jsx', size: 743, percentage: '0.06%' },
        { filename: 'Form.jsx', size: 2008, percentage: '0.17%' },
        { filename: 'List.jsx', size: 3910, percentage: '0.33%' },
        { filename: 'ListItem.jsx', size: 792, percentage: '0.07%' },
      ],
      [
        'client/stylesheets',
        [{ filename: 'styles.css', size: 453, percentage: '0.04%' }],
      ],
      [
        'node_modules/css-loader/dist/cjs.js!./node_modules/sass-loader/dist/cjs.js!./client/stylesheets',
        {
          filename: 'styles.css',
          size: 1293,
          percentage: '0.11%',
        },
      ],
      [
        'node_modules/css-loader/dist/runtime',
        { filename: 'api.js', size: 2677, percentage: '0.23%' },
      ],
    ],
    getBytes(number) {
      if (number < 1000) return `${number} B`;
      if (number < 1000000) return `${(number / 1000).toFixed(2)} KiB`;
      return `${(number / 1000000).toFixed(2)} MiB`;
    },
  };

  beforeEach(() => {
    wrapper = shallow(<ChangesTable {...props} />);
  });

  it('ChangesTable should render', () => {
    expect(wrapper);
  });

  test('ChangesTable snapshot testing', () => {
    expect(toJson(ChangesTable)).toMatchSnapshot();
  });

  it('ChangesTable should contain a div with classes cards-container and centered', () => {
    expect(wrapper.find('div').length).toEqual(1);
  });

  it('ChangesTable should render a SimpleExpansionPanel in the div', () => {
    expect(wrapper.find('SimpleExpansionPanel').length).toEqual(1);
  });
});
