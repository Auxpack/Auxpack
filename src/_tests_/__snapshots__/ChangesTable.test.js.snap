// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ChangesTable Unit Tests ChangesTable snapshot testing 1`] = `undefined`;
